src/segwit.rs

```diff
@@ -364,6 +378,8 @@ pub enum EncodeError {
     WitnessVersion(InvalidWitnessVersionError),
     /// Invalid witness length.
     WitnessLength(WitnessLengthError),
+    /// Encoding HRP, witver, and program into a bech32 string exceeds limit of 90 characters.
+    TooLong(EncodedLengthError),
```

Comment:
Why not `EncodedLengh` here?

Reply:
You were reviewing while I coded so this comment is stale now since I removed the error type all togther. FTR the policy we have come up with is to always prefix error types with `Error`. 

Anyways, please see the new version for better handling of the maximum string length.